compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/ExternalDependenciesGenerator.kt

```diff
@@ -34,19 +34,21 @@ class ExternalDependenciesGenerator(val symbolTable: SymbolTable, private val ir
         require(symbolTable.unboundTypeParameters.isEmpty()) { "Unbound type parameters are forbidden" }
 
         lateinit var unbound: List<IrSymbol>
-        fun <T> haveNotStabilized(prev: List<T>, cur: List<T>) =
-            cur.isNotEmpty() && (prev.size != cur.size || prev.any { !cur.contains(it) })
+        var notStabilized: Boolean
         do {
             unbound = symbolTable.allUnbound
+            notStabilized = false
 
             for (symbol in unbound) {
                 // Symbol could get bound as a side effect of deserializing other symbols.
                 if (!symbol.isBound) {
                     irProviders.getDeclaration(symbol)
                 }
+                // If any symbol became bound, the entire loop is not stabilized yet.
+                notStabilized = notStabilized || symbol.isBound
                 assert(symbol.isBound) { "${symbol.descriptor} unbound even after deserialization attempt" }
```

Comment:
Is there a legitimate case where `symbol` does not get bound?
It seems that the change only makes sense when the assertion is disabled.